src/arc.rs

```diff
@@ -183,47 +183,62 @@ impl PathArc {
     /// [`canonicalize`]: struct.PathAbs.html#method.canonicalize
     /// [`current_dir`]: fn.current_dir.html
     pub fn absolute(&self) -> Result<PathAbs> {
-        let mut components = self.components();
-        let mut stack: Vec<OsString> = Vec::new();
-
-        macro_rules! pop_stack { [] => {{
-            if let None = stack.pop() {
-                return Err(Error::new(
-                    io::Error::new(io::ErrorKind::NotFound, ".. consumed root"),
-                    "resolving absolute",
-                    self.clone(),
-                ));
-            }
-        }}}
-
-        handle_prefix(self, &mut stack, &mut components, false)?;
+        let mut res = current_dir(self)?.canonicalize()?;
+
+        for each in self.components() {
+            match each {
+                // This path starts with a (possibly already canonical) prefix
+                Component::Prefix(p) => {
+                    if p.kind().is_verbatim() {
+                        // This prefix already uses the extended-length
+                        // syntax, so we can use it as-is.
+                        res = PathAbs(res.join(p.as_os_str()));
```

Comment:
I'm confused. `res` starts as essentially `current_dir()`, but the first component of `self` is absolute... why are we doing `<current_dir()>.join(p)` here?

Reply:
so my confusing is when you do the first run through here.

if `self` is `/foo` and `res` is `/home/user` then won't the first thing this does be `res = PathAbs(current_dir().join("/"))` or am I reading it wrong?